src/iscsi.c

```diff
@@ -2027,8 +2038,6 @@ isboot_scsi_io(struct cam_sim *sim, union ccb *ccb)
 	int error;
 
 	ISBOOT_TRACE("isboot scsi io\n");
-	sess = (struct isboot_sess *)cam_sim_softc(sim);
-	mtx_unlock(&sess->cam_mtx);
```

Comment:
CAM does not require SIM lock for 4 years since 401ed17ad0ce7b3b248aa1eba7a091d08d30726c and FreeBSD 10. If this driver does not need it, it would be good to remove it completely, passing NULL to cam_sim_alloc().

Reply:
@amotin are you saying we could get rid of _all_ calls to mtx_lock and mtx_unlock in this code and do away with the "struct mtx cam_mtx" altogether?